### DISCUSSION TITLE
Error flashing Olimex BB-SMT32WL (STM32WLE5CCU6) via OpenOCD

### DISCUSSION THREAD

**Discussion by muhkuhgh** [stm32duino/Arduino_Core_STM32#2146] [Q&A]
Hey there,

I have an Olimex BB-SMT32WL Dev Kit based on the STM32WLE5CCU6.

I first tried to compile and flash the example using CubeIDE. It failed. I could fix compiling.. but I couldn't fix flashing.

I then opened a brand new empty CubeIDE STM32WLE5CCU6 project and simply enabled GPIO to set some pins high.. it compiled AND flashed (and run perfectly fine) using OpenOCD.. but as soon as I enable LoRaWAN, flashing fails.

So I switched to Arduino.. installed latest STM32Duino and stuff.. with LoRaWAN Lib I hade some undefined PC3 4 5 errors that I could fix by renaming them to PC_3..  compiled fine. Tried to flash .hex or .elf using OpenOCD.. Failed with the same error as before.. 

I then compiled an empty script.. but even the empty program throws me the error I'm getting using CubeIDE+LoRaWAN..

Maybe a RAM/FLASH Adress problem? But comparing the values I, as a newb, couldn't figure out what's causing the problem.

Any help very appreciated!

`Open On-Chip Debugger 0.12.0-00024-gc3ff39f8b (2023-06-26-16:51)

swd

adapter speed: 1000 kHz

 

stm32wlx_cti_prepare_restart_one

Info : Listening on port 6666 for tcl connections

Info : Listening on port 4444 for telnet connections

Info : CMSIS-DAP: SWD supported

Info : CMSIS-DAP: FW Version = 1.0

Info : CMSIS-DAP: Interface Initialised (SWD)

Info : SWCLK/TCK = 1 SWDIO/TMS = 1 TDI = 0 TDO = 0 nTRST = 0 nRESET = 1

Info : CMSIS-DAP: Interface ready

Info : clock speed 4000 kHz

Info : SWD DPIDR 0x6ba02477

Info : [stm32wlx.cm4] Cortex-M4 r0p1 processor detected

Info : [stm32wlx.cm4] target has 6 breakpoints, 4 watchpoints

Info : starting gdb server for stm32wlx.cm4 on 3333

Info : Listening on port 3333 for gdb connections

Info : accepting 'gdb' connection on tcp/3333

Info : device idcode = 0x10016497 (STM32WLE/WL5x - Rev 'unknown' : 0x1001)

Info : RDP level 0 (0xAA)

Info : flash size = 256 KiB

Info : flash mode : single-bank

undefined debug reason 8 - target needs reset

[stm32wlx.cm4] halted due to debug-request, current mode: Thread

xPSR: 0x01000000 pc: 0x1fff2b6c msp: 0x20001508

Info : Padding image section 0 at 0x0800f894 with 4 bytes (bank write end alignment)

Info : SWD DPIDR 0x6ba02477

Error: Failed to write memory at 0x20009ff8

Error: block write failed

Error: error writing to flash at address 0x08000000 at offset 0x00000000

Warn : keep_alive() was not invoked in the 1000 ms timelimit. GDB alive packet not sent! (6795 ms). Workaround: increase "set remotetimeout" in GDB

[stm32wlx.cm4] halted due to debug-request, current mode: Thread

xPSR: 0x01000000 pc: 0x1fff2b6c msp: 0x20001508

shutdown command invoked

Info : dropped 'gdb' connection`

**Comment by fpistm** [comment]
Hi @muhkuhgh 
don't know why you request support here. We do not support flash using openOCD. Try st community or wait answer to your issue open to openOCD GH.

**Reply by muhkuhgh** [reply]
Well, probably because flashing a compiled binary works in principle (as it does for a simple CubeIDE program) so it don't have to be a problem regarding flashing per se.. but could be as well a problem of wrong compiling..
And if you guys think it might be a problem due to the flashing procedure per se.. why not suggest another method..

**Reply by fpistm** [reply]
Well; as you have the issue with Cube IDE it is not link to STM32duino.
Maybe the flash is damaged and depending of the size of the flash used by your binary you are able to flash or not. Try with the STM32CubeProgrammer and select verify but have no idea and will not investigate more as I stated openOCD is not part of STM32 core upload method. That's all. ST Community is the right place to ask support else I will spend time to support other stuff than the one related to STM32duino which required already a huge support time.
